throttling/memory_gcra_test.go

```diff
@@ -0,0 +1,35 @@
+package throttling
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMemoryGCRA(t *testing.T) {
+	t.Run("burst and cost greater than rate", func(t *testing.T) {
+		l := &gcra{}
+
+		burst := int64(5)
+		rate := int64(1)
+		period := int64(1)
+
+		limit, err := l.limit(context.Background(), "key", burst+rate, burst, rate, period)
+		require.NoError(t, err)
+		require.True(t, limit, "it should be able to fill the bucket (burst)")
+
+		// next request should be allowed after 4 seconds
+		start := time.Now()
+		var allowed bool
+
+		require.Eventually(t, func() bool {
+			allowed, err = l.limit(context.Background(), "key", burst, burst, rate, period)
+			require.NoError(t, err)
+			return allowed
+		}, 10*time.Second, 1*time.Second, "next request should be eventually allowed")
+
+		require.GreaterOrEqual(t, time.Since(start).Seconds(), 5.0, "next request should be allowed after 5 seconds")
```

Comment:
If we do yet another request it should pass after 1 second, right? Should we add another assertion to make sure?

Reply:
The next request with a cost of 5 would pass after 5 seconds. This test asserts that costs higher than rate are eventually accepted, as long as they are not greater than the burst+rate

Reply:
Yes, I misread the test. The 3rd parameter is the `cost` and in the 1st call you're doing `burst+rate` so it makes perfect sense :+1: 

There might be another scenario which is splitting the 1st call into 5 and calling them in a row and then verify that:
1. the first 5 calls can be done without delay in between them (due to the burst setting)
2. eventually another call should be doable after >5s

Does it make sense?

Reply:
It was not the purpose of this test to verify the gcra algorithm itself as this is implemented by `throttled` external package. 
The purpose of this specific test scenario is to verify the that our cache expiration issue is indeed fixed.